src/plugin.rs

```diff
@@ -0,0 +1,69 @@
+use libloading::Library;
+use once_cell::sync::OnceCell;
+use std::ffi::{CString, OsStr};
+use std::ptr;
+use tree_sitter::Node;
+
+#[cfg(unix)]
+type RawSymbol<TValue> = libloading::os::unix::Symbol<TValue>;
+#[cfg(windows)]
+type RawSymbol<TValue> = libloading::os::windows::Symbol<TValue>;
```

Comment:
I'd find this slightly more clear if `TValue` was `TCFunction` or something like that.

Reply:
I think technically `TCFunction` is too "narrow" because exported symbols can also be things like globals, but yes `TValue` is a terrible name so I updated to `TExportedSymbol`